src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+                        AchIncomeTransaction achIncomeTransaction = new AchIncomeTransaction();
+                        achIncomeTransaction.setPaymentMethodCode(achIncomeFileTransaction.getPaymentMethodCode());
+                        achIncomeTransaction.setEffectiveDate(achIncomeFileTransaction.getEffectiveDate());
+                        achIncomeTransaction.setLoadTimestamp(dateTimeService.getCurrentTimestamp());
+                        achIncomeTransaction.setBankTimestamp(bankTimestamp);
+                        achIncomeTransaction.setTransactionAmount(achIncomeFileTransaction.getTransactionAmount());
+                        achIncomeTransaction.setPayerName(truncatePayerNameIfNecessary(achIncomeFileTransaction));
+                        achIncomeTransaction.setNotes(createNotes(achIncomeFileTransaction));
+                        achIncomeTransactions.add(achIncomeTransaction);
+                    }
+                }
+            }
+        }
+    }
+
+    protected String truncatePayerNameIfNecessary(AchIncomeFileTransaction achIncomeFileTransaction) {
+        String payerName = achIncomeFileTransaction.getPayerName();
+
+        if (payerName.length() > CuFPConstants.AchIncomeFileTransactionPayerOrPayeeName.ACH_TRN_PAYR_NM_DB_SIZE) {
+            payerName = payerName.substring(0, CuFPConstants.AchIncomeFileTransactionPayerOrPayeeName.ACH_TRN_PAYR_NM_DB_SIZE);
+        }
+
+        return payerName;
+    }
+
+    private List<AchIncomeNote> createNotes(AchIncomeFileTransaction achIncomeFileTransaction) {
+        StringBuffer notes = addPaymentInformationToNotes(achIncomeFileTransaction);
+        List<String> achNotes = addRMRAndNTELinesToNotes(achIncomeFileTransaction, notes);
+
+        List<AchIncomeNote> achIncomeNotes = new ArrayList<>();
+        int noteLineNumber = 1;
+        for (String noteText : achNotes) {
+            AchIncomeNote achIncomeNote = new AchIncomeNote();
+            achIncomeNote.setNoteLineNumber(noteLineNumber);
+            achIncomeNote.setNoteText(noteText);
+            achIncomeNotes.add(achIncomeNote);
+            noteLineNumber++;
+        }
+
+        return achIncomeNotes;
+    }
+
+    private StringBuffer addPaymentInformationToNotes(AchIncomeFileTransaction achIncomeFileTransaction) {
+        StringBuffer notes = new StringBuffer();
+
+        if (CuFPConstants.AchIncomeFileTransaction.TRANS_PAYMENT_METHOD_ACH.equals(achIncomeFileTransaction.getPaymentMethodCode())) {
+            notes.append("ACH PAYMENT INFORMATION: ");
+        }
+
+        if (CuFPConstants.AchIncomeFileTransaction.TRANS_PAYMENT_METHOD_FWT.equals(achIncomeFileTransaction.getPaymentMethodCode())) {
+            notes.append("WIRE PAYMENT INFORMATION: ");
+        }
+        notes.append("\n");
+        notes.append("CREDIT: ");
+        notes.append("$" + getFormattedAmount("##,##,###.00", achIncomeFileTransaction.getTransactionAmount()) + " " + achIncomeFileTransaction.getCreditDebitIndicator() + "\n");
```

Comment:
Would it be better to use "append" instead of "+" for building the string here and on other code lines below, given that it's using a builder/buffer anyway?